From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Wahbe <ian@wahbe.com>
Date: Mon, 6 Mar 2023 13:39:15 +0100
Subject: [PATCH] Add `shim.NewProvider` to expose `internal/provider.New`


diff --git a/shim/shim.go b/shim/shim.go
new file mode 100644
index 0000000..fb00354
--- /dev/null
+++ b/shim/shim.go
@@ -0,0 +1,10 @@
+package shim
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-docker/internal/provider"
+)
+
+func NewProvider() *schema.Provider {
+	return provider.New("dev")()
+}
