From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron Friel <mayreply@aaronfriel.com>
Date: Sun, 12 Mar 2023 19:00:46 -0700
Subject: [PATCH] Smarter default Docker client behavior


diff --git a/internal/provider/config.go b/internal/provider/config.go
index 27bc58d..f900a1d 100644
--- a/internal/provider/config.go
+++ b/internal/provider/config.go
@@ -126,6 +126,9 @@ func (c *ProviderConfig) MakeClient(ctx context.Context, d *schema.ResourceData)
 		return cached.(*client.Client), nil
 	}
 
+	opts := []client.Opt{
+		client.FromEnv,
+	}
 	if config.Cert != "" || config.Key != "" {
 		if config.Cert == "" || config.Key == "" {
 			return nil, fmt.Errorf("cert_material, and key_material must be specified")
@@ -140,29 +143,13 @@ func (c *ProviderConfig) MakeClient(ctx context.Context, d *schema.ResourceData)
 			return nil, err
 		}
 
-		// Note: don't change the order here, because the custom client
-		// needs to be set first them we overwrite the other options: host, version
-		dockerClient, err = client.NewClientWithOpts(
-			client.WithHTTPClient(httpClient),
-			client.WithHost(config.Host),
-			client.WithAPIVersionNegotiation(),
-		)
-		if err != nil {
-			return nil, err
-		}
+		opts = append(opts, client.WithHTTPClient(httpClient))
 	} else if config.CertPath != "" {
 		// If there is cert information, load it and use it.
 		ca := filepath.Join(config.CertPath, "ca.pem")
 		cert := filepath.Join(config.CertPath, "cert.pem")
 		key := filepath.Join(config.CertPath, "key.pem")
-		dockerClient, err = client.NewClientWithOpts(
-			client.WithHost(config.Host),
-			client.WithTLSClientConfig(ca, cert, key),
-			client.WithAPIVersionNegotiation(),
-		)
-		if err != nil {
-			return nil, err
-		}
+		opts = append(opts, client.WithTLSClientConfig(ca, cert, key))
 	} else if strings.HasPrefix(config.Host, "ssh://") {
 		// If there is no cert information, then check for ssh://
 		helper, err := connhelper.GetConnectionHelperWithSSHOpts(config.Host, config.SSHOpts)
@@ -170,22 +157,19 @@ func (c *ProviderConfig) MakeClient(ctx context.Context, d *schema.ResourceData)
 			return nil, err
 		}
 		if helper != nil {
-			dockerClient, err = client.NewClientWithOpts(
-				client.WithHost(helper.Host),
-				client.WithDialContext(helper.Dialer),
-				client.WithAPIVersionNegotiation(),
-			)
-			if err != nil {
-				return nil, err
-			}
+			opts = append(opts, client.WithHost(helper.Host), client.WithDialContext(helper.Dialer))
 		}
 	} else {
 		// If there is no ssh://, then just return the direct client
-		dockerClient, err = client.NewClientWithOpts(
-			client.WithHost(config.Host),
-			client.WithAPIVersionNegotiation(),
-		)
+		if config.Host != "" {
+			opts = append(opts, client.WithHost(config.Host))
+		}
 	}
+
+	opts = append(opts,
+		client.WithAPIVersionNegotiation(),
+	)
+	dockerClient, err = client.NewClientWithOpts(opts...)
 	if err != nil {
 		return nil, err
 	}
