From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: guineveresaenger <guinevere@pulumi.com>
Date: Tue, 31 Mar 2026 16:11:32 -0700
Subject: [PATCH] do not create empty TLS config


diff --git a/internal/provider/resource_docker_registry_image_funcs.go b/internal/provider/resource_docker_registry_image_funcs.go
index e0f5221..5c74433 100644
--- a/internal/provider/resource_docker_registry_image_funcs.go
+++ b/internal/provider/resource_docker_registry_image_funcs.go
@@ -296,15 +296,12 @@ func getAuthConfigForRegistry(
 }
 
 func buildHttpClientForRegistry(registryAddressWithProtocol string, insecureSkipVerify bool) *http.Client {
-	client := http.DefaultClient
-
-	if strings.HasPrefix(registryAddressWithProtocol, "https://") {
-		client.Transport = &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: insecureSkipVerify}, Proxy: http.ProxyFromEnvironment}
-	} else {
-		client.Transport = &http.Transport{Proxy: http.ProxyFromEnvironment}
+	if insecureSkipVerify && strings.HasPrefix(registryAddressWithProtocol, "https://") {
+		return &http.Client{
+			Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, Proxy: http.ProxyFromEnvironment},
+		}
 	}
-
-	return client
+	return http.DefaultClient
 }
 
 func deleteDockerRegistryImage(pushOpts internalPushImageOptions, registryWithProtocol string, sha256Digest, username, password string, insecureSkipVerify, fallback bool) error {
