From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bryce Lampe <bryce@pulumi.com>
Date: Wed, 21 May 2025 09:49:43 -0700
Subject: [PATCH] Remove service default


diff --git a/docs/resources/service.md b/docs/resources/service.md
index 24aa8fa..ef4f805 100644
--- a/docs/resources/service.md
+++ b/docs/resources/service.md
@@ -700,7 +700,7 @@ Required:
 
 Optional:
 
-- `global` (Boolean) The global service mode. Defaults to `false`
+- `global` (Boolean) When `true`, tasks will run on every worker node. Conflicts with `replicated`
 - `replicated` (Block List, Max: 1) The replicated service mode (see [below for nested schema](#nestedblock--mode--replicated))
 
 <a id="nestedblock--mode--replicated"></a>
diff --git a/internal/provider/resource_docker_service.go b/internal/provider/resource_docker_service.go
index 8b80e32..9a1433a 100644
--- a/internal/provider/resource_docker_service.go
+++ b/internal/provider/resource_docker_service.go
@@ -776,8 +776,7 @@ func resourceDockerService() *schema.Resource {
 						},
 						"global": {
 							Type:          schema.TypeBool,
-							Description:   "The global service mode. Defaults to `false`",
-							Default:       false,
+							Description:   "When `true`, tasks will run on every worker node. Conflicts with `replicated`",
 							Optional:      true,
 							ConflictsWith: []string{"mode.0.replicated", "converge_config"},
 						},
